src/containers/Settings/EmailPreferences.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import styles from './Settings.module.scss';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { map as fpMap } from 'lodash';
+import { PageCard, FormSection } from 'explorer-components';
+import { Field, reduxForm } from 'redux-form';
+import validators from 'utils/validators';
+import { FormToggle } from './components';
+import { Cropper, Button, Text } from 'components';
+import { actions as settingsActions } from 'public-modules/Settings';
+import { emailPreferencesSelector } from 'public-modules/Settings/selectors';
+import { getCurrentUserSelector } from 'public-modules/Authentication/selectors';
+import { EMAIL_NOTIFICATION_OPTIONS } from './constants';
+
+const map = fpMap.convert({ cap: false });
+
+let EmailPreferencesComponent = props => {
+  const {
+    error,
+    saving,
+    submitFailed,
+    invalid,
+    onSubmit,
+    handleSubmit,
+    saveEmailPreferences
+  } = props;
+
+  const handleSaveEmailPreferences = values => {
+    saveEmailPreferences(values);
+  };
+
+  const renderToggles = () => {
+    return map(
+      (value, key) => (
+        <div className={`col-xs-12 ${styles.emailToggle}`}>
+          <Field
+            form={'emailPreferences'}
+            disabled={saving}
+            name={key}
+            component={FormToggle}
+            label={value}
+          />
+        </div>
+      ),
+      EMAIL_NOTIFICATION_OPTIONS
+    );
+  };
+
+  return (
+    <form onSubmit={handleSubmit(handleSaveEmailPreferences)}>
+      <FormSection>
+        <FormSection.Section title="EMAIL NOTIFICATIONS">
+          <FormSection.Description>
+            What notifications would you like to receive via email?
+          </FormSection.Description>
+          <FormSection.SubText>
+            Opt in or out of the notifications you wish to receive via email. We
+            recommend sticking with the default settings so that you can be
+            informed of important activity relevant to you on the network.
+          </FormSection.SubText>
+          <FormSection.InputGroup>
+            <div className="row">{renderToggles()}</div>
+          </FormSection.InputGroup>
+        </FormSection.Section>
+      </FormSection>
+      <PageCard.Break />
+      <div className={styles.buttons}>
+        <Button
+          type="primary"
+          disabled={saving || (submitFailed && invalid)}
+          loading={saving}
+        >
+          Save Email Preferences
+        </Button>
+        {submitFailed && invalid ? (
+          <Text inputLabel color="red" className={styles.submitError}>
+            Fix errors before submitting.
+          </Text>
+        ) : null}
+      </div>
+    </form>
+  );
+};
+
+const mapStateToProps = state => {
+  const prefs = getCurrentUserSelector(state).settings.emails;
```

Comment:
haha can you just call this preferences? :)

Reply:
😁